packages/flutter/lib/src/material/dialog.dart

```diff
@@ -73,13 +89,19 @@ class Dialog extends StatelessWidget {
   /// {@endtemplate}
   final ShapeBorder shape;
 
-  Color _getColor(BuildContext context) {
-    return Theme.of(context).dialogBackgroundColor;
+  /// The widget below this widget in the tree.
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  Color _getColor(BuildContext context, DialogTheme dialogTheme) {
+    return backgroundColor ?? dialogTheme.backgroundColor ?? Theme.of(context).dialogBackgroundColor;
```

Comment:
Since this function is only used once, it would be OK to just inline it. Doing so would highlight the similarity between the Material parameter values. Note also: wide lines are OK.